### ISSUE TITLE
Trains broke with NaN value

### ISSUE THREAD

**Issue by lamhoangtung** [allegroai/trains#118]
Hi. Recently my experiment throwing some NaN value and make the whole thing brake since Trains can't handle it.

```python
Traceback (most recent call last):
  File "train.py", line 129, in <module>
    launch_train_with_config(traincfg, trainer)
  File "/usr/local/lib/python3.6/dist-packages/tensorpack/train/interface.py", line 99, in launch_train_with_config
    extra_callbacks=config.extra_callbacks)
  File "/usr/local/lib/python3.6/dist-packages/tensorpack/train/base.py", line 342, in train_with_defaults
    steps_per_epoch, starting_epoch, max_epoch)
  File "/usr/local/lib/python3.6/dist-packages/tensorpack/train/base.py", line 314, in train
    self.main_loop(steps_per_epoch, starting_epoch, max_epoch)
  File "/usr/local/lib/python3.6/dist-packages/tensorpack/utils/argtools.py", line 168, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorpack/train/base.py", line 279, in main_loop
    self.run_step()  # implemented by subclass
  File "/usr/local/lib/python3.6/dist-packages/tensorpack/train/base.py", line 179, in run_step
    self.hooked_sess.run(self.train_op)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/training/monitored_session.py", line 754, in run
    run_metadata=run_metadata)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/training/monitored_session.py", line 1252, in run
    run_metadata=run_metadata)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/training/monitored_session.py", line 1353, in run
    raise six.reraise(*original_exc_info)
  File "/usr/lib/python3/dist-packages/six.py", line 693, in reraise
    raise value
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/training/monitored_session.py", line 1338, in run
    return self._sess.run(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/training/monitored_session.py", line 1419, in run
    run_metadata=run_metadata))
  File "/usr/local/lib/python3.6/dist-packages/tensorpack/callbacks/hooks.py", line 36, in after_run
    self._cb.after_run(ctx, vals)
  File "/usr/local/lib/python3.6/dist-packages/tensorpack/callbacks/base.py", line 133, in after_run
    self._after_run(run_context, run_values)
  File "/usr/local/lib/python3.6/dist-packages/tensorpack/callbacks/summary.py", line 117, in _after_run
    self.trainer.monitors.put_summary(summary)
  File "/usr/local/lib/python3.6/dist-packages/tensorpack/callbacks/monitor.py", line 161, in put_summary
    self._dispatch(lambda m: m.process_summary(summary))
  File "/usr/local/lib/python3.6/dist-packages/tensorpack/callbacks/monitor.py", line 138, in _dispatch
    func(m)
  File "/usr/local/lib/python3.6/dist-packages/tensorpack/callbacks/monitor.py", line 161, in <lambda>
    self._dispatch(lambda m: m.process_summary(summary))
  File "/usr/local/lib/python3.6/dist-packages/tensorpack/callbacks/monitor.py", line 274, in process_summary
    self._writer.add_summary(summary, self.global_step)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/summary/writer/writer.py", line 142, in add_summary
    self._add_event(event, global_step)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/summary/writer/writer.py", line 392, in _add_event
    super(FileWriter, self)._add_event(event, step)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/summary/writer/writer.py", line 279, in _add_event
    self.event_writer.add_event(event)
  File "/usr/local/lib/python3.6/dist-packages/trains/binding/frameworks/tensorflow_bind.py", line 629, in add_event
    ret = ev.add_event(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/trains/binding/frameworks/tensorflow_bind.py", line 519, in add_event
    self._add_scalar(tag=tag, step=step, scalar_data=values)
  File "/usr/local/lib/python3.6/dist-packages/trains/binding/frameworks/tensorflow_bind.py", line 290, in _add_scalar
    self._scalar_report_cache[(title, series)] = (num + 1, value + scalar_data)
TypeError: unsupported operand type(s) for +: 'int' and 'str'
```
Turn out `scalar_data` in the trace above is a `NaN` value when I tried to inspect it. When I turn trains off. Everything still work since Tensorboard can handle `NaN` value.

I hope this will be fix soon. Really enjoy using Trains so far. Thanks for the amazing works !

**Comment by bmartinn** [comment]
Hi @lamhoangtung nice catch !
I still wonder how you got the NaN in the first place ;)

Anyhow, please see if the latest RC fixes the problem:
```bash
pip install trains==0.14.1rc1
```

**Comment by lamhoangtung** [comment]
Lol my dumb loss implementation keep giving me NaN result nah. My bad.

Anw, thanks for the quick fix @bmartinn. I will do some test and get back to you soon ;)

**Comment by allegroai-git** [comment]
Hi @lamhoangtung,

This fix is included in the newly released Trains v0.14.1
```bash
$ pip install trains==0.14.1
```

**Comment by bmartinn** [comment]
Hi @lamhoangtung,

This fix is included in the newly released Trains v0.14.1 :)
```bash
$ pip install trains==0.14.1
```

**Comment by lamhoangtung** [comment]
@bmartinn Thanks. The problem was resolved for me
